deploy/charts/csi-driver/templates/daemonset.yaml

```diff
@@ -17,7 +17,7 @@ spec:
       containers:
 
         - name: node-driver-registrar
-          image: "{{ .Values.nodeDriverRegistrarImage.repository }}:{{ .Values.nodeDriverRegistrarImage.tag }}"
+          image: "{{ .Values.nodeDriverRegistrarImage.repository }}{{- if (.Values.nodeDriverRegistrarImage.digest) -}} @{{.Values.nodeDriverRegistrarImage.digest}}{{- else -}}:{{ default  .Values.nodeDriverRegistrarImage.tag }} {{- end -}}"
```

Comment:
@MattiasGees it looks like you have included a whitespace before the `@` symbol here, which I think is unintentional. There's also a whitespace before the `{{- end -}}` at the very end of the line too.

I'm also uncertain, but I think having that final `-}}` will cause the templator to 'eat' the trailing newline afterwards, and therefore produce invalid YAML. Have you tested this out (with both a tag & with a digest) using something like `helm template` to verify?

Reply:
The format that @MattiasGees proposes is very similar to the format used in cert-manager: https://github.com/cert-manager/cert-manager/blob/master/deploy/charts/cert-manager/templates/webhook-deployment.yaml#L60.